src/XamlX.IL.Cecil/Resolvers/GenericParameterResolver.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace XamlX.TypeSystem;
+
+// Yet another Cecil member that was never exposed nor used outside of WinMD code
+// https://github.com/jbevain/cecil/blob/56d4409b8a0165830565c6e3f96f41bead2c418b/Mono.Cecil/GenericParameterResolver.cs#L5
+
+internal sealed class GenericParameterResolver
+{
+	internal static TypeReference ResolveReturnTypeIfNeeded(MethodReference methodReference)
+	{
+		if (methodReference.DeclaringType.IsArray && methodReference.Name == "Get")
+			return methodReference.ReturnType;
+
+		var genericInstanceMethod = methodReference as GenericInstanceMethod;
+		var declaringGenericInstanceType = methodReference.DeclaringType as GenericInstanceType;
+
+		if (genericInstanceMethod == null && declaringGenericInstanceType == null)
+			return methodReference.ReturnType;
+
+		return ResolveIfNeeded(genericInstanceMethod, declaringGenericInstanceType, methodReference.ReturnType);
+	}
+
+	internal static TypeReference ResolveFieldTypeIfNeeded(FieldReference fieldReference)
+	{
+		return ResolveIfNeeded(null, fieldReference.DeclaringType as GenericInstanceType, fieldReference.FieldType);
+	}
+
+	internal static TypeReference ResolvePropertyTypeIfNeeded(PropertyReference propertyReference)
+	{
+		return ResolveIfNeeded(null, propertyReference.DeclaringType as GenericInstanceType, propertyReference.PropertyType);
+	}
+
+	internal static TypeReference ResolveParameterTypeIfNeeded(MethodReference method, ParameterReference parameter)
+	{
+		var genericInstanceMethod = method as GenericInstanceMethod;
+		var declaringGenericInstanceType = method.DeclaringType as GenericInstanceType;
+
+		if (genericInstanceMethod == null && declaringGenericInstanceType == null)
+			return parameter.ParameterType;
+
+		return ResolveIfNeeded(genericInstanceMethod, declaringGenericInstanceType, parameter.ParameterType);
+	}
+
+	internal static TypeReference ResolveParameterTypeIfNeeded(PropertyReference property, ParameterReference parameter)
+	{
+		var declaringGenericInstanceType = property.DeclaringType as GenericInstanceType;
+
+		if (declaringGenericInstanceType == null)
+			return parameter.ParameterType;
+
+		return ResolveIfNeeded(null, declaringGenericInstanceType, parameter.ParameterType);
+	}
```

Comment:
This method is pretty much the only addition from me here, as property related methods were missing in the original code.

Reply:
For the original code, see file headers.